labrad/test/test_fasttypes.py

```diff
@@ -8,14 +8,20 @@ def test_int():
     assert ft.flatten(4, 'i') == ('\x00\x00\x00\x04', 'i')
 def test_str():
     assert ft.flatten('abcd', 's') == ('\x00\x00\x00\x04abcd', 's')
-def test_wrong_type():
+def test_wrong_type(): # Figure out the right exceptions to use here
     with pytest.raises(Exception):
         ft.flatten('abcd', 'i')
     with pytest.raises(Exception):
         ft.flatten(4, 's')
 def test_list():
     assert all(T.unflatten(*ft.flatten([1,2,3])) == [1,2,3])
-    assert all(ft.unflatten(T.flatten([1,2,3])) == [1,2,3])
+    (data, tt) = T.flatten([1,2,3])
+    data, tt = str(data), str(tt)
```

Comment:
fasttypes only works with strings
types.py returns LRType objects and sometimes uses Buffer()'s for the data.

On Thu, Apr 23, 2015 at 2:11 PM, Matthew Neeley <notifications@github.com>
wrote:

> In labrad/test/test_fasttypes.py
> <https://github.com/martinisgroup/pylabrad/pull/91#discussion_r29003971>:
>
> >      with pytest.raises(Exception):
> >          ft.flatten('abcd', 'i')
> >      with pytest.raises(Exception):
> >          ft.flatten(4, 's')
> >  def test_list():
> >      assert all(T.unflatten(*ft.flatten([1,2,3])) == [1,2,3])
> > -    assert all(ft.unflatten(T.flatten([1,2,3])) == [1,2,3])
> > +    (data, tt) = T.flatten([1,2,3])
> > +    data, tt = str(data), str(tt)
>
> why do you have to call str on these? what is the actual return type?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/martinisgroup/pylabrad/pull/91/files#r29003971>.
>